src/screens/SendSynthetic/SendSyntheticAmount.js

```diff
@@ -0,0 +1,384 @@
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import * as React from 'react';
+import { Keyboard, TouchableOpacity } from 'react-native';
+import { NavigationScreenProp } from 'react-navigation';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { SDK_PROVIDER } from 'react-native-dotenv';
+import styled from 'styled-components/native';
+import t from 'tcomb-form-native';
+import get from 'lodash.get';
+import isEmpty from 'lodash.isempty';
+import debounce from 'lodash.debounce';
+
+// components
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import Button from 'components/Button';
+import Spinner from 'components/Spinner';
+import { BaseText, Label, TextLink } from 'components/Typography';
+import { Wrapper } from 'components/Layout';
+
+// actions
+import { initSyntheticsServiceAction } from 'actions/syntheticsActions';
+
+// utils, services
+import { fontStyles, spacing, UIColors } from 'utils/variables';
+import { formatAmount, formatFiat, isValidNumber, parseNumber } from 'utils/common';
+import { getAssetData, getAssetsAsList, getRate } from 'utils/assets';
+import syntheticsService from 'services/synthetics';
+import { getAmountFormFields } from 'utils/formHelpers';
+
+// constants
+import { defaultFiatCurrency, PLR } from 'constants/assetsConstants';
+import { SEND_SYNTHETIC_CONFIRM, SEND_TOKEN_CONFIRM } from 'constants/navigationConstants';
+
+// selectors
+import { accountAssetsSelector } from 'selectors/assets';
+
+// models, types
+import type { Asset, Assets, Rates } from 'models/Asset';
+import type { SyntheticTransaction, TokenTransactionPayload } from 'models/Transaction';
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
+
+type Props = {
+  accountAssets: Assets,
+  supportedAssets: Asset[],
+  initSyntheticsService: () => void,
+  navigation: NavigationScreenProp<*>,
+  rates: Rates,
+  baseFiatCurrency: ?string,
+  isOnline: boolean,
+};
+
+type State = {
+  value: ?{ amount: ?string },
+  submitPressed: boolean,
+  intentError: ?string,
+  inputHasError: boolean,
+};
+
+const { Form } = t.form;
+
+const generateFormStructure = (intentError: ?string, maxAmount: number, decimals: number) => {
+  const Amount = t.refinement(t.String, (amount): boolean => {
+    if (!isValidNumber(amount.toString()) || intentError) return false;
+
+    if (decimals === 0 && amount.toString().includes('.')) return false;
+    amount = parseFloat(amount.toString());
+
+    return (decimals !== 0 || !amount.toString().includes('.')) && amount <= maxAmount;
+  });
+
+  Amount.getValidationErrorMessage = (amount) => {
+    if (!isValidNumber(amount.toString())) {
+      return 'Incorrect number entered.';
+    }
+
+    amount = parseNumber(amount.toString());
+
+    if (amount > maxAmount) {
+      return 'Amount should not exceed the max available';
+    } else if (decimals === 0 && amount.toString().includes('.')) {
```

Comment:
Maybe we should extract this to separate function like `validateNumberDecimals`?